firmware/console/status_loop.cpp

```diff
@@ -430,8 +430,9 @@ static void updateFuelSensors() {
 
 	engine->outputChannels.flexPercent = Sensor::getOrZero(SensorType::FuelEthanolPercent);
 #if EFI_PROD_CODE
-	// todo: extract method? do better?
+	// todo: extract method? do better? see https://github.com/rusefi/rusefi/issues/7511 for details
 	engine->module<InjectorModelSecondary>()->pressureCorrectionReference = engine->module<InjectorModelSecondary>()->getFuelDifferentialPressure().Value + Sensor::get(SensorType::Map).value_or(STD_ATMOSPHERE);
```

Comment:
if you don't have an SD card and/or TS connected, pressureCorrectionReference is never updated, so this feature is broken

Reply:
Damn I've even asked myself that question yesterday but made wrong conclusion :( just added extra documentation.

Thank you!

Reply:
in general, two code smells that are good to eliminate:

1. any "load bearing" logic in `status_loop.cpp`
2. anything with the form of `engine->module<MyModule>()->myModuleMember = x`. Do it instead inside a member function of `MyModule`.

Reply:
right, between mocks/overrides I am often afraid to touch these complex classes

note how https://github.com/rusefi/rusefi/commit/fb8f4486f9966296244ef975cecb67ece68cafeb was not a success requiring https://github.com/rusefi/rusefi/commit/ed4bf68c237793e24c2f2883ff8b2cf11c162095 follow-up